src/json_export.cpp

```diff
@@ -51,33 +61,41 @@ JsonExporter::ExpectedEntry JsonExporter::fromJson(const nlohmann::json& source)
   {
     return nonstd::make_unexpected("json object is null");
   }
-  if(source.is_string())
-  {
-    return Entry{ BT::Any(source.get<std::string>()),
-                  BT::TypeInfo::Create<std::string>() };
-  }
-  if(source.is_number_unsigned())
-  {
-    return Entry{ BT::Any(source.get<uint64_t>()), BT::TypeInfo::Create<uint64_t>() };
-  }
-  if(source.is_number_integer())
-  {
-    return Entry{ BT::Any(source.get<int64_t>()), BT::TypeInfo::Create<int64_t>() };
-  }
-  if(source.is_number_float())
-  {
-    return Entry{ BT::Any(source.get<double>()), BT::TypeInfo::Create<double>() };
-  }
-  if(source.is_boolean())
+  if(!source.contains("__type"))
```

Comment:
I believe I saw that `nlohmann::json` has a `find` function.

```suggestion
  auto source_type_it = source.find(kTypeField);
  if(source_type_it == source.end())
```

Then you can replace usage of `source[kTypeField]` with `source_type_it` below.

Reply:
Looks like the only other usage of kTypeField in this function is to lookup if the type is in the available converters (`type_names_`) so I wouldn't be able to reuse this iterator.